hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -114,9 +113,82 @@ public class RocksDBCheckpointDiffer {
   private static final long UNKNOWN_COMPACTION_GEN = 0;
   private long currentCompactionGen = 0;
 
-  // Something to track all the snapshots created so far.
+  // Something to track all the snapshots created so far. TODO: REMOVE
   private Snapshot[] allSnapshots;
 
+  private String compactionLogParentDir = null;
+  private String compactionLogDir = null;
+
+  // Name of the directory to hold compaction logs (under parent dir)
+  private static final String COMPACTION_LOG_DIR = "compaction-log/";
+
+  // For DB compaction SST DAG persistence and reconstruction
+  // Should be initialized to the latest sequence number
+  private volatile String currentCompactionLogFilename = null;
+
+  private static final String COMPACTION_LOG_FILENAME_SUFFIX = ".log";
+
+  public void setCompactionLogParentDir(String parentDir) {
+    this.compactionLogParentDir = parentDir;
+
+    // Append /
+    if (!compactionLogParentDir.endsWith("/")) {
+      compactionLogParentDir += "/";
+    }
+
+    File pDir = new File(compactionLogParentDir);
+    if (!pDir.exists()) {
+      if (!pDir.mkdir()) {
+        LOG.error("Error creating compaction log parent dir.");
+        return;
+      }
+    }
+
+    compactionLogDir =
+        Paths.get(compactionLogParentDir, COMPACTION_LOG_DIR).toString();
+    File clDir = new File(compactionLogDir);
+    if (!clDir.mkdir()) {
+      LOG.error("Error creating compaction log dir.");
+      return;
+    }
+
+    // TODO: Write a README there explaining what the dir is for
+  }
+
+  private static final int LONG_MAX_STRLEN =
+      String.valueOf(Long.MAX_VALUE).length();
+
+  public void setCompactionLogFilenameBySeqNum(long latestSequenceId) {
+    String latestSequenceIdStr = String.valueOf(latestSequenceId);
+
+    if (latestSequenceIdStr.length() < LONG_MAX_STRLEN) {
+      // Pad zeroes to the left for sequential listing later
+      latestSequenceIdStr =
+          StringUtils.leftPad(latestSequenceIdStr, LONG_MAX_STRLEN, "0");
+    }
+
+    this.currentCompactionLogFilename = compactionLogParentDir +
+        COMPACTION_LOG_DIR + latestSequenceIdStr +
+        COMPACTION_LOG_FILENAME_SUFFIX;
+
+    // Create empty file when if it doesn't exist.
+    // Detect any write permission issue (if any) early.
+    File clFile = new File(currentCompactionLogFilename);
+    if (clFile.exists()) {
+      LOG.warn("Compaction log exists: {}. Will append",
```

Comment:
It seems like things will break, if appendToCurrentCompactionLog() gets called right here between the if statement and the else statement.